src/api/runner.ts

```diff
@@ -63,31 +80,37 @@ export class TestRunner extends vscode.Disposable {
 
         const tests = request.include || [];
         const root = this._workspaceFolder.uri.fsPath;
+        const settings = this._context.workspaceState.get<SettingOptions>("tpn.settings");
+
+        const files = getTestFiles(tests);
+        const opts = getRunTestsOpts(tests, settings);
 
         if (!tests.length) {
             const browserItems = this._tree.getBrowserTestItemsByFolderPath(root);
             browserItems.forEach(browserItem => this._enqueueTestItem(browserItem));
 
             logger.info(`Running all tests in ${basename(root)} folder`);
-            await this._api.runTests([]);
-
-            return;
+        } else {
+            tests.forEach(testItem => this._enqueueTestItem(testItem));
+
+            if (!_.isEmpty(opts)) {
+                logger.info(`Running ${files.length} file(s) with options: ${JSON.stringify(opts)}`);
+            } else {
+                logger.info(
+                    `Running ${files.length} file(s):`,
+                    files.map(f => relative(root, f)),
+                );
+            }
         }
 
-        tests.forEach(testItem => this._enqueueTestItem(testItem));
-        const files = getTestFiles(tests);
-        const opts = getRunTestsOpts(tests);
+        try {
+            await this._api.runTests(files, opts);
+        } catch (err) {
+            this._testRun?.end();
+            this._testRun = undefined;
 
-        if (!_.isEmpty(opts)) {
-            logger.info(`Running ${files.length} file(s) with options: ${JSON.stringify(opts)}`);
-        } else {
-            logger.info(
-                `Running ${files.length} file(s):`,
-                files.map(f => relative(root, f)),
-            );
+            throw err;
```

Comment:
send error to main `extension.ts` file in order to show popup with error for user.